platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -190,8 +230,19 @@ - (void)startWithQueue:(dispatch_queue_t)queue completionHandler:(MGLMapSnapshot
                     compositedImage = [[NSImage alloc] initWithSize:targetSize];
                     
                     [compositedImage lockFocus];
+
                     [sourceImageRep drawInRect: targetFrame];
-                    [logoImage drawInRect:CGRectMake(MGLLogoImagePosition.x, MGLLogoImagePosition.y, logoImage.size.width,logoImage.size.height)];
+                    
+                    CGRect logoImageRect = CGRectMake(MGLLogoImagePosition.x, MGLLogoImagePosition.y, logoImage.size.width, logoImage.size.height);
+                    [logoImage drawInRect:logoImageRect];
+                    
+                    CGPoint attributionTextPosition = CGPointMake(targetFrame.size.width - 10 - attributionBackgroundSize.width,
+                                                                  logoImageRect.origin.y + (logoImageRect.size.height / 2) - (attributionBackgroundSize.height / 2));
```

Comment:
As tail work, we’ll have to figure out how to shrink attribution in narrower images, especially when there are multiple attribution strings, as in Mapbox Satellite Streets. Possibilities include replacing the logo image with one that omits the wordmark and also abbreviating some commonly attributed names.